src/pages/index.js

```diff
@@ -42,54 +33,175 @@ export function handleImageClick(name, link) {
-const cardList = new Section(
-  { data: initialCards, renderer: renderCard },
-  ".cards__list",
-  handleImageClick
-);
-cardList.renderItems();
+// Like Card Functionality
+function handleLikeClick(card) {
+  if (!card._isLiked) {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .removeLikeCard(card._id)
+      .then(() => {
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 
-// enabling validation by calling enableValidation()
-const editFormValidator = new FormValidator(config, profileEditForm);
-const addFormValidator = new FormValidator(config, addCardForm);
-addFormValidator.enableValidation();
-editFormValidator.enableValidation();
+const createCard = (cardObject) => {
+  const card = new Card(
+    cardObject,
+    "#cards__list-item",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+  return card.getView();
+};
 
 // enable userInfo
 const userInfo = new UserInfo({
   userName: ".profile__title",
-  userJob: ".profile__description",
+  userAbout: ".profile__description",
+  userAvatar: ".profile__avatar",
 });
 
-// enablePopupWithForm on editProfile
-const editProfilePopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleEditProfileSubmit
-);
-editProfilePopup.setEventListeners();
+// enable API calls
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "a93ebd2f-fd06-499f-b2f9-74e2f8519709",
+    "Content-Type": "application/json",
+  },
+});
 
-// enablePopupWithForm on newCard
-const newCardPopup = new PopupWithForm(
-  "#profile-add-modal",
-  handleNewCardSubmit
-);
-newCardPopup.setEventListeners();
+// Render user info and cards on page load
+Promise.all([api.getUser(), api.getInitialCards()]).then(([data, cards]) => {
+  userInfo.setUserInfo({ name: data.name, about: data.about });
+  userInfo.setUserAvatar(data.avatar, data.name);
+  cardList = new Section(
+    { data: cards, renderer: renderCard },
+    ".cards__list",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+  cardList.renderItems();
+});
 
-// enable PopupWithImage on cards
-const cardPopup = new PopupWithImage("#card-modal");
-cardPopup.setEventListeners();
+// enabling validation by calling enableValidation()
+const editFormValidator = new FormValidator(config, profileEditForm);
+editFormValidator.enableValidation();
+
+const addFormValidator = new FormValidator(config, addCardForm);
+addFormValidator.enableValidation();
+
+const avatarFormValidator = new FormValidator(config, avatarForm);
+avatarFormValidator.enableValidation();
+
+// Editing Profile Interaction
+// Edit Profile Information
+const editProfilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
+  editProfilePopup.setSubmitText(true);
```

Comment:
The code for rendering loading is correct, you do everything in the right places